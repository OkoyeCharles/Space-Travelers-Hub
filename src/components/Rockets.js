import React, { useEffect } from 'react';
import { useDispatch } from 'react-redux';
import fetchRockets from '../redux/rocket/actions';

function Rockets() {
  // const rockets = useSelector((state) => state.rockets);
  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(fetchRockets());
  }, []);

  return (
    <div className="rocket__container">
      <div className="col m-3 p-3 border">
        column to display rockets
      </div>
    </div>
  );
}

export default Rockets;
