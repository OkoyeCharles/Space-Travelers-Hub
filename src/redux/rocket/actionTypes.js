export const GET_ROCKETS = 'store/rockets/GET_ROCKETS';
export const RESERVE_ROCKET = 'store/rockets/RESERVE_ROCKET';
