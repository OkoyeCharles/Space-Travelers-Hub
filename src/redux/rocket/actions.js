import axios from 'axios';
import * as actionType from './actionTypes';

const BASE_URL = 'https://api.spacexdata.com/v3/rockets';

const loadRockets = (rocketData) => {
  const rockets = rocketData.map((rocket) => ({
    id: rocket.id,
    rocketId: rocket.rocket_id,
    name: rocket.rocket_name,
    type: rocket.rocket_type,
    image: rocket.flickr_images[0],
    description: rocket.description,
  }));
  return {
    type: actionType.GET_ROCKETS,
    payload: rockets,
  };
};

const fetchRockets = () => async (dispatch) => {
  await axios({
    method: 'get',
    url: BASE_URL,
    responseType: 'json',
  })
    .then((res) => {
      dispatch(loadRockets(res.data));
    });
};

export default fetchRockets;
