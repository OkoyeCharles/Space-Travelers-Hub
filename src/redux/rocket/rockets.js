import * as actionType from './actionTypes';

const initialState = [];

const rocketsReducer = (state = initialState, action) => {
  switch (action.type) {
    case actionType.GET_ROCKETS:
      return [...action.payload];
    default:
      return state;
  }
};

export default rocketsReducer;
