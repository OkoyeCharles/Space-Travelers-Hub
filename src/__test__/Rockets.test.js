import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import Rockets from '../components/Rockets';
import store from '../redux/configureStore';
import { reserveRocket, loadRockets } from '../redux/rocket/actions';

describe('Missions tests', () => {
  it('render test', () => {
    const tree = renderer.create(
      <Provider store={store}>
        <Rockets />
      </Provider>,
    );

    expect(tree).toMatchSnapshot();
  });

  it('should return an action with type GET_ROCKETS', () => {
    const data = [
      {
        id: 1,
        rocket_id: 'rock52',
        rocket_name: 'rock 52',
        rocket_type: 'rocket',
        flickr_images: ['image1.png', 'imag2.jpg'],
        description: 'A test rocket',
      },
    ];
    const action = loadRockets(data);
    expect(action.type).toBe('store/rockets/GET_ROCKETS');
  });

  it('should return an action with type RESERVE_ROCKET', () => {
    const id = 1;
    const action = reserveRocket(id);
    expect(action.type).toBe('store/rockets/RESERVE_ROCKET');
  });
});
