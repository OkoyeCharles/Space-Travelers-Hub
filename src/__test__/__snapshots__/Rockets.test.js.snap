// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Missions tests render test 1`] = `
<div
  className="rocket__container"
>
  <div
    className="col m-3 p-3 border"
  />
</div>
`;
